### PR TITLE
qa: [Auctions2] QA items 

### PR DESCRIPTION
**Pull request by damassi**
Addresses: 
- https://artsyproduct.atlassian.net/browse/GRO-275
- https://artsyproduct.atlassian.net/browse/GRO-276
- https://artsyproduct.atlassian.net/browse/GRO-274
- https://artsyproduct.atlassian.net/browse/GRO-273
- https://artsyproduct.atlassian.net/browse/GRO-263

Fixes: 
- Exclude bidding-closed works from "Works by artists you follow" rail
- Seeing Attempted Load For MyBids, for user with no bids.
- Update Headers for "Active Bids and Watched Lots" and "Works By Artists You Follow"
- Ensure that WorksByArtistsYouFollow is hidden if user follows no artists
- User sees zero state on /auctions tabs when no auctions under filter

Also adds analytics tracking tests.

### CODE REVIEW

File: src/v2/Apps/Auctions/AuctionsApp.tsx [hunk 1]
```diff
@@ -12,20 +12,20 @@ import { RecentlyViewedQueryRenderer as RecentlyViewed } from "v2/Components/Rec
 import { RouterLink } from "v2/Artsy/Router/RouterLink"
 import { RouteTabs, RouteTab } from "v2/Components/RouteTabs"
 import { useSystemContext } from "v2/Artsy"
-import { WorksByArtistsYouFollowRailQueryRenderer } from "./Components/WorksByArtistsYouFollowRail/WorksByArtistsYouFollowRail"
+import { WorksByArtistsYouFollowRailFragmentContainer } from "./Components/WorksByArtistsYouFollowRail/WorksByArtistsYouFollowRail"
 export interface AuctionsAppProps {
-  me: AuctionsApp_me
+  viewer: AuctionsApp_viewer
```
Comment (The-Beez-Kneez) [inline src/v2/Apps/Auctions/AuctionsApp.tsx:17]: Thank you! This had been bugging me so badly.

**Review by The-Beez-Kneez — COMMENTED** [review]
This looks good, I would request that we add in some tests for the new zero state in `Current/Upcoming/Past` Auctions.

But otherwise looks good.

File: src/v2/Apps/Auctions/Components/MyBids/MyBidsBidHeader.tsx [hunk 2]
```diff
@@ -38,6 +38,7 @@ export const MyBidsBidHeader: React.FC<MyBidsBidHeaderProps> = ({ sale }) => {
             style={{ objectFit: "cover" }}
             src={sale.coverImage.resized.src}
             srcSet={sale.coverImage.resized.srcSet}
+            lazyLoad
```
Comment (damassi) [inline src/v2/Apps/Auctions/Components/MyBids/MyBidsBidHeader.tsx:41]: @The-Beez-Kneez - this simple prop on our `Image` component is how you make an image fade in when the image scrolls into view. Its super helpful for performance.

File: src/v2/Apps/Auctions/Components/AuctionArtworksRail/__tests__/AuctionArtworksRail.jest.tsx [hunk 3]
```diff
@@ -45,4 +61,17 @@ describe.skip("AuctionArtworksRail", () => {
...
+    wrapper.find("RouterLink").first().simulate("click")
+    expect(trackEvent).toHaveBeenCalledWith({
+      action: "clickedArtworkGroup",
+      context_module: "currentAuctions",
+      destination_page_owner_id: '<mock-value-for-field-"internalID">',
+      destination_page_owner_slug: '<mock-value-for-field-"slug">',
+      destination_page_owner_type: "sale",
+      type: "viewAll",
+    })
+  })
```
Comment (damassi) [inline src/v2/Apps/Auctions/Components/AuctionArtworksRail/__tests__/AuctionArtworksRail.jest.tsx:76]: @artsy/grow-devs - Not sure if y'all are familiar with this strategy, but once the in general tracking setup is added to the test file (mocks etc), and tracking added to the file under test, its easy to find  the values here by leaning on jest's diffing function: 

Steps
1) 
```tsx
expect(trackEvent).toHaveBeenCalledWith('hey now') 
```
2) Run test, check jest's diff
```
 Expected: "hey now"
    Received: {"action": "clickedArtworkGroup", "context_module": "worksByArtistsYouFollowRail", "context_page_owner_id": undefined, "context_page_owner_slug": undefined, "context_page_owner_type": undefined, "destination_page_owner_id": "<mock-value-for-field-\"internalID\">", "destination_page_owner_slug": "<mock-value-for-field-\"slug\">", "destination_page_owner_type": "artwork", "horizontal_slide_position": 0, "type": "thumbnail"}
```
3) Copy that received section into test and delete the props that are undefined:
```tsx
  expect(trackEvent).toHaveBeenCalledWith({
      action: "clickedArtworkGroup",
      context_module: "currentAuctions",
      destination_page_owner_id: '<mock-value-for-field-"internalID">',
      destination_page_owner_slug: '<mock-value-for-field-"slug">',
      destination_page_owner_type: "sale",
      type: "viewAll",
  })
    ```
4) Verify that it looks right by comparing against the file being tested. 

Note that this jest-leaning can be done with all kinds of things.

File: src/v2/Components/Artwork/FillwidthItem.tsx [hunk 4]
```diff
@@ -53,12 +53,18 @@ export class FillwidthItemContainer extends React.Component<
...
       artwork: {
         image: { aspectRatio },
       },
     } = this.props
 
+    // Our relay testing layer often mocks props with strings; this makes it a
+    // little safer and less confusing when that happens.
+    if (typeof aspectRatio !== "number") {
+      aspectRatio = 1
+    }
```
Comment (damassi) [inline src/v2/Components/Artwork/FillwidthItem.tsx:66]: We could mock out `aspectRatio` every time in jest tests, but that's no fun.

**Comment by damassi** [comment]
(Ready for another look y'all)

**Review by The-Beez-Kneez — APPROVED** [review]

**Comment by The-Beez-Kneez** [comment]
@damassi looks ready to merge to me, thanks for adding in those additional tests.

File: src/v2/Apps/Auctions/__tests__/AuctionsApp.jest.tsx [hunk 5]
```diff
@@ -83,4 +99,21 @@ describe("AuctionsApp", () => {
     expect(html).not.toContain("Ended")
     expect(html).not.toContain("In Progress")
   })
+
+  it("does not render MyBids or WorksByFollowedArtists if user logged out", () => {
+    useSystemContext.mockImplementation(() => ({
+      user: null,
```
Comment (jonallured) [inline src/v2/Apps/Auctions/__tests__/AuctionsApp.jest.tsx:105]: **question [non-blocking]:** passing in the user

I see how you're assembling this test case to have a `null` user, but I'm missing how other test cases have a user?? The `mockImplementation` call in the `beforeEach` above only specifies `on` and `off` - where do the other tests get a user from?

*format via [Conventional Comments](https://conventionalcomments.org/)*

**Review by jonallured — APPROVED** [review]
Nothing blocking but a couple Qs for you and one thought about bailing on rendering that rail sooner. Really nice work here!! 🏆

File: src/v2/Apps/Auctions/Components/AuctionArtworksRail/__tests__/AuctionArtworksRail.jest.tsx [hunk 6]
```diff
@@ -25,6 +26,21 @@ describe.skip("AuctionArtworksRail", () => {
...
+  beforeEach(() => {
+    useTracking.mockImplementation(() => {
+      return {
+        trackEvent,
+      }
+    })
+  })
+
+  afterEach(() => {
+    trackEvent.mockReset()
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/v2/Apps/Auctions/Components/AuctionArtworksRail/__tests__/AuctionArtworksRail.jest.tsx:41]: Often times this is copy 🍝 from patterns in other files - but I'm with you, I prefer to use `clearAllMocks`, will update 👍

File: src/v2/Apps/Auctions/Components/WorksByArtistsYouFollowRail/WorksByArtistsYouFollowRail.tsx [hunk 7]
```diff
@@ -28,6 +21,10 @@ const WorksByArtistsYouFollowRail: React.FC<WorksByArtistsYouFollowRailProps> =
   const { contextPageOwnerType } = useAnalyticsContext()
   const contextModule = tabTypeToContextModuleMap.worksByArtistsYouFollow
 
+  if (viewer.saleArtworksConnection.edges.length === 0) {
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/v2/Apps/Auctions/Components/WorksByArtistsYouFollowRail/WorksByArtistsYouFollowRail.tsx:24]: All good points! but....

Bailing higher up: Because of Relay data masking, data for a component needs to be resolved within that component. I mean, one could def define a fully-resolved schema up above, but then we'd be defining it twice while _also_ breaking component-data encapsulation patterns.

Moving to top of file: One would think, yah? Unfortunately that breaks the [rule of hooks](https://www.benmvp.com/blog/conditional-react-hooks). Its some React magic that i'm not too fond of, given that pre-hooks React didn't have much of that.

[the remark this answers is not in the document]
Comment (jonallured) [reply src/v2/Apps/Auctions/Components/WorksByArtistsYouFollowRail/WorksByArtistsYouFollowRail.tsx:24]: Darn, oh well!! 😝

**Comment by damassi** [comment]
@The-Beez-Kneez - before clicking merge, good to check to see if there's any other new comments that might indicate that its still being worked on (I was [doing an update](https://github.com/artsy/force/pull/7291#discussion_r607993128) based on Johns feedback. And in any case, once PR is approved and has 🍏,  the dev can merge.

**Comment by The-Beez-Kneez** [comment]
Sorry, will do next time!
